### DISCUSSION TITLE
Included reference is not present in returned record and therefore causes a TypeScript error

### DISCUSSION THREAD

**Discussion by manuelpoelzl** [prisma/prisma#14733] [Q&A] [answered]
### Bug description

When I include a reference in a query, the referenced record is not present in the result, which results in a TypeScript error because the property does not exist.

The model:
```javascript
model Artifact {

  ID             Int               @id @unique @default(autoincrement())
  Name           String            @unique @map("name") @db.VarChar(200)
  AssignedSite   ExcavationSite    @relation(fields: [AssignedSiteID], references: [ID], onDelete: Cascade)
  AssignedSiteID Int
  Notes          String            @default("No notes available") @map("notes") @db.VarChar(255)
  Owner          User              @relation(fields: [OwnerID], references: [ID])
  OwnerID        Int               @map("owner")
  Image          String?           @map("assignedImage") @db.VarChar(32)
  InTrash        Boolean           @default(false) @map("inTrash")
  DiscoveryDate  DateTime?         @map("discoveryDate") @db.Date
  Category       ArtifactCategory? @relation(fields: [CategoryID], references: [ID], onDelete: SetNull)
  CategoryID     Int               @map("category")

  @@map("artifacts")
}
```
The query:
```typescript
const matchingArtifact: Artifact = await client.artifact.findFirstOrThrow(
                {where: {Owner: user, ID: id, InTrash: true},include:{AssignedSite: true}});
```
The error that occurs then:
![image](https://user-images.githubusercontent.com/46539758/183831035-230147ed-48f2-4dd6-9a1e-523f309a2a2e.png)

### How to reproduce

1. Create a model with a reference
2. Include this reference in a query
3. Try to use the included reference in TypeScript

### Expected behavior

The included reference should be available and usable in the result

### Prisma information

Model:
```javascript
model Artifact {

  ID             Int               @id @unique @default(autoincrement())
  Name           String            @unique @map("name") @db.VarChar(200)
  AssignedSite   ExcavationSite    @relation(fields: [AssignedSiteID], references: [ID], onDelete: Cascade)
  AssignedSiteID Int
  Notes          String            @default("No notes available") @map("notes") @db.VarChar(255)
  Owner          User              @relation(fields: [OwnerID], references: [ID])
  OwnerID        Int               @map("owner")
  Image          String?           @map("assignedImage") @db.VarChar(32)
  InTrash        Boolean           @default(false) @map("inTrash")
  DiscoveryDate  DateTime?         @map("discoveryDate") @db.Date
  Category       ArtifactCategory? @relation(fields: [CategoryID], references: [ID], onDelete: SetNull)
  CategoryID     Int               @map("category")

  @@map("artifacts")
}
```
Query:

```typescript
const matchingArtifact: Artifact = await client.artifact.findFirstOrThrow(
                {where: {Owner: user, ID: id, InTrash: true},include:{AssignedSite: true}});
```

### Environment & setup

- OS: Windows 10 21H1
- Database: MySql
- Node.js version: 16.13.1

### Prisma Version

```
prisma                  : 4.1.1
@prisma/client          : 4.1.1
Current platform        : windows
Query Engine (Node-API) : libquery-engine 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at node_modules\@prisma\engines\query_engine-windows.dll.node)      
Migration Engine        : migration-engine-cli 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at node_modules\@prisma\engines\migration-engine-windows.exe)  
Introspection Engine    : introspection-core 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at node_modules\@prisma\engines\introspection-engine-windows.exe)
Format Binary           : prisma-fmt 8d8414deb360336e4698a65aa45a1fbaf1ce13d8 (at node_modules\@prisma\engines\prisma-fmt-windows.exe)
Default Engines Hash    : 8d8414deb360336e4698a65aa45a1fbaf1ce13d8
Studio                  : 0.469.0

```

**Comment by austincrim** [comment, marked as answer]
Hey @manuelpoelzl 👋,

Remove the `: Artifact` type annotation and let TypeScript inference provide the type information for the variable. 

`const matchingArtifact = ...`

Prisma's types are set up correctly to include/exclude relation types depending on the query itself.

**Reply by manuelpoelzl** [reply]
Wow, i didn´t know that this works out of the box without having to assign a type!

Thank you very much for clearing this up!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
